src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/NamingPolicyUnitTests.cs

```diff
@@ -0,0 +1,199 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class NamingPolicyUnitTests
+    {
+        [Fact]
+        public static void ToCamelCaseTest()
+        {
+            // These test cases were copied from Json.NET.
+            Assert.Equal("urlValue", Convert("URLValue"));
+            Assert.Equal("url", Convert("URL"));
+            Assert.Equal("id", Convert("ID"));
+            Assert.Equal("i", Convert("I"));
+            Assert.Equal("", Convert(""));
+            Assert.Null(Convert(null));
+            Assert.Equal("person", Convert("Person"));
+            Assert.Equal("iPhone", Convert("iPhone"));
+            Assert.Equal("iPhone", Convert("IPhone"));
+            Assert.Equal("i Phone", Convert("I Phone"));
+            Assert.Equal("i  Phone", Convert("I  Phone"));
+            Assert.Equal(" IPhone", Convert(" IPhone"));
+            Assert.Equal(" IPhone ", Convert(" IPhone "));
```

Comment:
`iPhone` sounds very appley. I don't care. We are now an open source community.
But how about using something more microsofty here?

Reply:
That's the standard test case which came from Newtonsoft's JSON.

Reply:
Fine :)